File: appUtil/src/main/java/com/theroyalsoft/mydoc/apputil/baseUI/RoundDashView.java
package com.theroyalsoft.mydoc.apputil.baseUI;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.Path.Direction;
import android.util.AttributeSet;
import android.view.View;

public class RoundDashView extends View {

    private RoundShape roundShape1;
    private RoundShape roundShape2;
    private RoundShape roundShape3;
    private RoundShape roundShape4;


    private float ratioRadius1;
    private float ratioRadius2;
    private float ratioRadius3;
    private float ratioRadius4;

    public RoundDashView(Context context) {
        super(context);
        initMyView();
    }

    public RoundDashView(Context context, AttributeSet attrs) {
        super(context, attrs);
        initMyView();
    }

    public RoundDashView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        initMyView();
    }

    public void initMyView() {
        roundShape1 = new RoundShape();
        roundShape2 = new RoundShape();
        roundShape3 = new RoundShape();
        roundShape4 = new RoundShape();
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);

        int w = getWidth();
        int h = getHeight();

        if ((w == 0) || (h == 0)) {
            return;
        }

        float x = (float) w / 2.0f;
        float y = (float) h / 2.0f;


        roundShape1.setCircle(x, y, ratioRadius1, Direction.CCW);
        roundShape2.setCircle(x, y, ratioRadius2, Direction.CCW);
        roundShape3.setCircle(x, y, ratioRadius3, Direction.CCW);
        roundShape4.setCircle(x, y, ratioRadius4, Direction.CCW);

        canvas.drawPath(roundShape1.getPath(), roundShape1.getPaint());
        canvas.drawPath(roundShape2.getPath(), roundShape2.getPaint());
        canvas.drawPath(roundShape3.getPath(), roundShape3.getPaint());
        canvas.drawPath(roundShape4.getPath(), roundShape4.getPaint());
    }


    public void setShapeRadiusRatio(float ratio1, float ratio2, float ratio3, float ratio4) {

        ratioRadius1 = ratio1;
        ratioRadius2 = ratio2;
        ratioRadius3 = ratio3;
        ratioRadius4 = ratio4;
    }


}

File: app/src/main/java/com/theroyalsoft/telefarmer/utils/RoundDashView.kt
package com.theroyalsoft.telefarmer.utils

import android.content.Context
import android.graphics.Canvas
import android.graphics.Path
import android.util.AttributeSet
import android.view.View

/**
 * Created by Pritom Dutta on 31/7/23.
 */
class RoundDashView : View {
    private var roundShape1: RoundShape? = null
    private var roundShape2: RoundShape? = null
    private var roundShape3: RoundShape? = null
    private var roundShape4: RoundShape? = null
    private var ratioRadius1 = 0f
    private var ratioRadius2 = 0f
    private var ratioRadius3 = 0f
    private var ratioRadius4 = 0f

    constructor(context: Context?) : super(context) {
        initMyView()
    }

    constructor(context: Context?, attrs: AttributeSet?) : super(context, attrs) {
        initMyView()
    }

    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(
        context,
        attrs,
        defStyleAttr
    ) {
        initMyView()
    }

    fun initMyView() {
        roundShape1 = RoundShape()
        roundShape2 = RoundShape()
        roundShape3 = RoundShape()
        roundShape4 = RoundShape()
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        val w = width
        val h = height
        if (w == 0 || h == 0) {
            return
        }
        val x = w.toFloat() / 2.0f
        val y = h.toFloat() / 2.0f
        roundShape1!!.setCircle(x, y, ratioRadius1, Path.Direction.CCW)
        roundShape2!!.setCircle(x, y, ratioRadius2, Path.Direction.CCW)
        roundShape3!!.setCircle(x, y, ratioRadius3, Path.Direction.CCW)
        roundShape4!!.setCircle(x, y, ratioRadius4, Path.Direction.CCW)
        canvas.drawPath(roundShape1!!.getPath(), roundShape1!!.getPaint())
        canvas.drawPath(roundShape2!!.getPath(), roundShape2!!.getPaint())
        canvas.drawPath(roundShape3!!.getPath(), roundShape3!!.getPaint())
        canvas.drawPath(roundShape4!!.getPath(), roundShape4!!.getPaint())
    }

    fun setShapeRadiusRatio(ratio1: Float, ratio2: Float, ratio3: Float, ratio4: Float) {
        ratioRadius1 = ratio1
        ratioRadius2 = ratio2
        ratioRadius3 = ratio3
        ratioRadius4 = ratio4
    }
}